asus_stylus.py

```diff
@@ -23,82 +25,97 @@
 
 tries = 5
 
 # Look into the devices file
 while tries > 0:
 
-    stylus_detected = 0
+    stylus_detection_status = 0
 
-    with open('/proc/bus/input/devices', 'r') as f:
-        lines = f.readlines()
+    with open('/proc/bus/input/devices', 'r') as devices_list_file:
+        lines = devices_list_file.readlines()
         for line in lines:
 
             # Look for the stylus
-            if stylus_detected == 0 and "Stylus" in line:
-                stylus_detected = 1
+            if stylus_detection_status == 0 and "Stylus" in line:
+                stylus_detection_status = 1
                 log.debug('Detect stylus from %s', line.strip())
 
-            if stylus_detected == 1:
+            # Found stylus, now searching for ids
+            if stylus_detection_status == 1:
                 if "S: " in line:
                     # search device id
-                    device_id=re.sub(r".*i2c-(\d+)/.*$", r'\1', line).replace("\n", "")
-                    log.debug('Set stylus device id %s from %s', device_id, line.strip())
+                    stylus_id = re.sub(r".*i2c-(\d+)/.*$", r'\1', line).replace("\n", "")
+                    device_ids.append(stylus_id)
+                    log.debug('Set stylus device id %s from %s', stylus_id, line.strip())
 
                 if "H: " in line:
-                    stylus = line.split("event")[1]
-                    stylus = stylus.split(" ")[0]
-                    stylus_detected = 2
-                    log.debug('Set stylus id %s from %s', stylus, line.strip())
+                    stylus_id = line.split("event")[1].split(" ")[0]
+                    styluses.append(stylus_id)
+                    log.debug('Set stylus id %s from %s', stylus_id, line.strip())
+                    stylus_detection_status = 0
 
-            # Stop looking if stylus have been found
-            if stylus_detected == 2:
-                break
+    # Stylus was found on at least one device
+    if len(styluses) > 0:
+        stylus_detection_status = 2;
 
-    if stylus_detected != 2:
+    # Stylus was not detected?
+    if stylus_detection_status != 2:
         tries -= 1
         if tries == 0:
-            if stylus_detected != 2:
-                log.error("Can't find stylus (code: %s)", stylus_detected)
-            if stylus_detected == 2 and not device_id.isnumeric():
-                log.error("Can't find device id")
+            if stylus_detection_status != 2:
+                log.error("Can't find stylus (code: %s)", stylus_detection_status)
+            for device_id in device_ids:
+                if stylus_detection_status == 2 and not device_id.isnumeric():
+                    log.error("Can't find device id")
             sys.exit(1)
     else:
         break
 
 
 # Start monitoring the stylus
-fd_t = open('/dev/input/event' + str(stylus), 'rb')
-d_t = Device(fd_t)
+stylus_devices = []
+for stylus in styluses:
+    file_io = open('/dev/input/event' + str(stylus), 'rb')
```

Comment:
@Safenein Could you please `file_io` name `fd` (file descriptor)

Reply:
Yes I can do that.

Reply:
Done in 0fc6a79.